test/SigningKeys.t.sol

```diff
@@ -0,0 +1,594 @@
+            startIndex,
+            keysCount,
+            pubkeys,
+            signatures
+        );
+    }
+
+    function test_saveKeysSigs_revertWhen_toManyKeys() public {
+        uint256 keysCount = type(uint32).max;
+        uint16 startIndex = 2;
+        uint256 nodeOperatorId = 154;
+
+        (bytes memory pubkeys, bytes memory signatures) = (
+            new bytes(0),
+            new bytes(0)
+        );
+
+        vm.expectRevert(SigningKeys.InvalidKeysCount.selector);
+        signingKeys.saveKeysSigs(
+            nodeOperatorId,
+            startIndex,
+            keysCount,
+            pubkeys,
+            signatures
+        );
+    }
+
+    function test_saveKeysSigs_revertWhen_invalidSigsLen() public {
+        uint256 keysCount = 10;
+        uint16 startIndex = 2;
+        uint256 nodeOperatorId = 154;
+        (bytes memory pubkeys, ) = keysSignatures(keysCount, startIndex);
+        vm.expectRevert(SigningKeys.InvalidLength.selector);
+        signingKeys.saveKeysSigs(
+            nodeOperatorId,
+            startIndex,
+            keysCount,
+            pubkeys,
+            new bytes(0)
+        );
+    }
+
+    function test_saveKeysSigs_revertWhen_invalidPubkeysLen() public {
+        uint256 keysCount = 10;
+        uint16 startIndex = 2;
+        uint256 nodeOperatorId = 154;
+        (, bytes memory signatures) = keysSignatures(keysCount, startIndex);
+        vm.expectRevert(SigningKeys.InvalidLength.selector);
+        signingKeys.saveKeysSigs(
+            nodeOperatorId,
+            startIndex,
+            keysCount,
+            new bytes(0),
+            signatures
+        );
+    }
+
+    function test_saveKeysSigs_revertWhen_EmptyKey() public {
+        uint256 keysCount = 10;
+        uint16 startIndex = 2;
+        uint256 nodeOperatorId = 154;
+        (
+            bytes memory pubkeys,
+            bytes memory signatures
+        ) = keysSignaturesWithZeroKey(keysCount, startIndex, 3);
+
+        vm.expectRevert(SigningKeys.EmptyKey.selector);
+        signingKeys.saveKeysSigs(
+            nodeOperatorId,
+            startIndex,
+            keysCount,
+            pubkeys,
+            signatures
+        );
+    }
+
+    function testFuzz_saveKeysSigs(
+        uint64 nodeOperatorId,
+        uint32 startIndex,
+        uint32 keysCount
+    ) public {
+        vm.assume(keysCount > 0);
+        vm.assume(keysCount < 100);
+        unchecked {
+            vm.assume(startIndex + keysCount > startIndex);
+        }
+        vm.assume(startIndex + keysCount < type(uint32).max);
+
+        (bytes memory pubkeys, bytes memory signatures) = keysSignatures(
+            keysCount,
+            uint16(startIndex)
+        );
+        uint256 newKeysCount = signingKeys.saveKeysSigs(
+            nodeOperatorId,
+            startIndex,
+            keysCount,
+            pubkeys,
+            signatures
+        );
+
+        assertEq(newKeysCount, startIndex + keysCount);
```

Comment:
I would add a check that `loadKeys` returns saved ones.

Reply:
added